modules/axp/Animal/lib/LivingCreature.pm

```diff
@@ -21,7 +24,19 @@ our $VERSION = '0.01';
 
 Don't use this class directly; instead instantiate subclasses of it.
 
-=head1 EXPORT
+=cut
+
+our $__constructor_check = sub {
+	ref ( shift ) and croak 'Static constructor used as instance call';
+};
+our $__static_check = sub {
+	ref ( shift ) and croak 'Class method used as instance call';
+};
+our $__instance_check = sub {
+	ref ( shift ) or croak 'Instance variable needed';
+};
+
+before 'new' => $__constructor_check;
```

Comment:
There's the hook on the 'new' constructor&mdash; stop people writing `$object->new` instead of `Class->new`.